os/fs/driver/mtd/smart.c

```diff
@@ -5634,12 +5642,13 @@ static int smart_journal_recovery(FAR struct smart_struct_s *dev, journal_log_t
 			return -EIO;
 		}
 
-		/* And then check validation of crc, if crc is matched, it means both of data & heade are valid,
-		 * so only checkout is necessary.. But Should we check written mtd header with journal log's ?
+		/* And then check validation of crc, if crc is matched, it means both of data & header are valid.
+		 * But Should we check written mtd header with journal log's?
 		 */
 		ret = smart_validate_crc(dev);
 		if (ret == OK) {
 			fvdbg("Entire of sector is valid, so checkout last journal!!!!\n");
+			/* Data dn MTD header were succesfully updated but Journal Entry was not checked out. Do it now */
```

Comment:
Please check `dn`. I think `Data in MTD header`.

Reply:
My bad, thankyou.